### DISCUSSION TITLE
CSS templates for export to PDF / Preview

### DISCUSSION THREAD

**Discussion by claudiorrrr** [notaapp/talk#141] [Feature Requests] [answered]
I'm most of all in love of nota. But there's one killer feature I can't live without that unfortunately I haven't found any information about: templates for exporting to PDF. 

The user case is you write, say, a report on the great Nota app and then you'd like to export it in a professional fashion, hopefully with some personality from your own. You can do that [using iaWriter](https://github.com/iainc/iA-Writer-Templates) (my favorite), [Ulysses](https://styles.ulysses.app/), and even [Marked2](https://marked2app.com/help/Custom_Styles.html). But unfortunately **you can't do that** on Nota. The best you have is the default template (not really beautiful to be honest ;)) at export to PDF.

It would be brilliant to have the chance to export to PDF with an open template structure. Is that something you have thought about?

**Comment by astoilkov** [comment, marked as answer]
We've thought about that. There are a lot of improvements that can be made around PDF exporting. I will directly share our internal thought process and the options we are currently considering:

- Baking in all the functionality as part of Nota
- Allowing an easy way to extend Nota (through extensibility) and implement them as you want. That is currently possible but we want to make it easier for people to share those extensions so other people like you can use them. We can even make some of the extensions ourselves
- Leave that type of functionality to Marked2 as it's a feature-rich app with lots of functionalities that we will hardly match

The conclusion is, we are thinking about this but we are still not sure about the best path forward. What do you think about the different options?

**Reply by claudiorrrr** [reply]
In my modest opinion, I would say between 2 and 3. 1 is interesting and will provide more value to the app, but the downside is that probably that will limit the flexibility needed. 

I'm not a dev so I don't know whether 2 and 3 are complementary, but if that's possible, that will be my preferred way. #2, providing an extension-based way to explore templates with advanced options (better than iawriter, for instance) would be delightful as well. Happy to help testing out.

**Reply by astoilkov** [reply]
Great. Thanks for sharing your thoughts.

**Comment by lolbat** [comment]
I would love to see Nota have a PDF export CSS file that we could edit. Also the ability to switch them in the export dialog? Then I could have export options per project that I am working on.
